### PR TITLE
Make event_type link table

### PR DESCRIPTION
**Pull request by svineet**
Melange task: http://www.google-melange.com/gci/task/view/google/gci2014/5234308872667136

### CODE REVIEW

File: modules/s3db/deploy.py [hunk 1]
```diff
@@ -332,6 +335,13 @@ def model(self):
                        deploy_assignment = "mission_id",
                        deploy_alert = "mission_id",
                        deploy_response = "mission_id",
+                       event_type={
```
Comment (hitesh96db) [inline modules/s3db/deploy.py]: deploy prefix is needed

File: modules/s3db/deploy.py [hunk 2]
```diff
@@ -1193,9 +1212,26 @@ def deploy_rheader(r, tabs=[], profile=False):
                                TH("%s: " % table[f].label, **attr)
                 value = lambda f, table=table, record=record, **attr: \
                                TD(table[f].represent(record[f]), **attr)
+
+                def event_type_link_label():
```
Comment (hitesh96db) [inline modules/s3db/deploy.py]: Writing functions like this inside, makes it hard to read(poor readability).
You could just save it in a variable

File: modules/s3db/deploy.py [hunk 3]
```diff
@@ -1193,9 +1212,26 @@ def deploy_rheader(r, tabs=[], profile=False):
...
+                    db = current.db
+                    return TH(db
+                              .deploy_mission_event_type["event_type_id"]
+                              .label)
+
+                def event_type_link_value():
+                    db = current.db
+                    row = db(
+                        db.deploy_mission_event_type.mission_id
```
Comment (hitesh96db) [inline modules/s3db/deploy.py]: db.deploy_mission_event_type can be saved in a variable.
Also, this indentation may follow pep8 rules, but this looks very odd ;). Look at the code around you.

File: modules/s3db/deploy.py [hunk 4]
```diff
@@ -1193,9 +1212,26 @@ def deploy_rheader(r, tabs=[], profile=False):
...
+                        == record.id).select().first()
```
Comment (hitesh96db) [inline modules/s3db/deploy.py]: Select only the things you need(i.e, only event_type_id) and you could put limitby = (0, 1). Saves an extra function call.

File: modules/s3db/deploy.py [hunk 5]
```diff
@@ -1193,9 +1212,26 @@ def deploy_rheader(r, tabs=[], profile=False):
...
+                    return TD(db
```
Comment (hitesh96db) [inline modules/s3db/deploy.py]: if it's getting too long, just save it in a variable, and use it

**Comment by svineet** [comment]
@hitesh96db I have done the changes required

File: modules/s3db/deploy.py [hunk 6]
```diff
@@ -1193,9 +1212,21 @@ def deploy_rheader(r, tabs=[], profile=False):
                                TH("%s: " % table[f].label, **attr)
                 value = lambda f, table=table, record=record, **attr: \
                                TD(table[f].represent(record[f]), **attr)
+
+                db = current.db
+                event_type_ltable = db.deploy_mission_event_type
```
Comment (hitesh96db) [inline modules/s3db/deploy.py]: ltable would be enough

File: modules/s3db/deploy.py [hunk 7]
```diff
@@ -1193,9 +1212,21 @@ def deploy_rheader(r, tabs=[], profile=False):
...
+                row = db(event_type_ltable.mission_id == record.id).select(
+                    event_type_ltable.event_type_id,
+                    limitby=(0, 1))[0]
+                event_type_link_value = TD(event_type_ltable.event_type_id
+                                           .represent(row["event_type_id"]))
+
                 rheader = DIV(H2(title),
-                              TABLE(TR(label("event_type_id"),
-                                       value("event_type_id"),
+                              TABLE(TR(event_type_link_label,
```
Comment (hitesh96db) [inline modules/s3db/deploy.py]: event_type_label ? Short and precise

File: modules/s3db/deploy.py [hunk 8]
```diff
@@ -1193,9 +1212,21 @@ def deploy_rheader(r, tabs=[], profile=False):
                                TH("%s: " % table[f].label, **attr)
                 value = lambda f, table=table, record=record, **attr: \
                                TD(table[f].represent(record[f]), **attr)
+
+                db = current.db
+                event_type_ltable = db.deploy_mission_event_type
+                event_type_link_label = TH(event_type_ltable["event_type_id"]
+                                           .label)
+
+                row = db(event_type_ltable.mission_id == record.id).select(
```
Comment (hitesh96db) [inline modules/s3db/deploy.py]: <pre>
row = db(ltable.missions_id == record.id).select(ltable.event_type_id,
                                                 limitby=(0, 1))[0]
</pre>

Comment (hitesh96db) [inline modules/s3db/deploy.py]: <pre>
db(long_query).select(table.field,
                      limitby=(0, 1))[0] 
</pre>

Comment (hitesh96db) [inline modules/s3db/deploy.py]: It may be under 80 with ltable instead of event_......

File: modules/s3db/deploy.py [hunk 9]
```diff
@@ -1193,9 +1212,19 @@ def deploy_rheader(r, tabs=[], profile=False):
                                TH("%s: " % table[f].label, **attr)
                 value = lambda f, table=table, record=record, **attr: \
                                TD(table[f].represent(record[f]), **attr)
+
+                db = current.db
+                ltable = db.deploy_mission_event_type
+                event_type_label = TH(ltable["event_type_id"].label)
+
+                event_id_table = ltable.event_type_id
```
Comment (hitesh96db) [inline modules/s3db/deploy.py]: Misleading name ? it's a field no ? ;)

**Pull request merged**